File: src/PiWebService/PiStatus.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.Json;
using System.Threading.Tasks;

namespace PiWebService
{
    public class PiStatus
    {
        public double Humidity { get; set; }

        public double Temp { get; set; }

        public double HeatIndex { get; set; }

        public double DewPoint { get; set; }

        public int ErrorsSinceLastUpdate { get; set; }

        public int TimeSinceLastUpdate { get; set; }

        public long Time { get; set; }

        public PiClockStatus[] Clocks { get; set; }
    }

    public class PiClockStatus
    {
        public string Name { get; set; }

        public string DisplayName { get; set; }

        public string Status { get; set; }

        public long TimeInSeconds { get; set; }

        internal void Update(PiClockStatus newClock) {
            Status = newClock.Status;
            TimeInSeconds = newClock.TimeInSeconds;
        }
    }

    public class PiCommand
    {
        public string Name { get; set; }

        public JsonDocument Data { get; set; }
    }
}

File: src/PiWebInterface/app/models/pi.status.ts

export interface PiClockStatus {
    name: string;
    displayName: string;
    status: string;
    timeInSeconds: number;
}

export interface PiCommand {
    name: string;
    data: any;
}

export interface PiStatus {
    temp: number;
    humidity: number;
    heatIndex: number;
    dewPoint: number;
    errorsSinceLastUpdate: number;
    time: number;
    timeSinceLastUpdate: number;
    clocks: PiClockStatus[];
}



